example/service/s3/getObjectWithProgress/getObjectWithProgress.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"regexp"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"strings"
+	"fmt"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"gopkg.in/cheggaaa/pb.v1"
+)
+
+type progressWriter struct {
+	w  io.WriterAt
+	pb *pb.ProgressBar
+}
+
+func (pw *progressWriter) WriteAt(p []byte, off int64) (int, error) {
+	pw.pb.Add(len(p))
+	return pw.w.WriteAt(p, off)
+}
+
+func byteCountDecimal(b int64) string {
+	const unit = 1000
+	if b < unit {
+		return fmt.Sprintf("%d B", b)
+	}
+	div, exp := int64(unit), 0
+	for n := b / unit; n >= unit; n /= unit {
+		div *= unit
+		exp++
+	}
+	return fmt.Sprintf("%.1f %cB", float64(b)/float64(div), "kMGTPE"[exp])
+}
+
+func getFileSize(svc *s3.S3, bucket string, prefix string) (filesize int64, error error) {
```

Comment:
I think using the [HeadObject](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#S3.HeadObject) API operation would be better instead of list object. This will also ensure that no other objects are returned, and additional error checking isn't needed.

Reply:
Nice, I was repurposing some of my existing `ls` code, but for a single files, makes sense to use `HeadObject`